core/capabilities/compute/compute.go

```diff
@@ -116,9 +123,47 @@ func (c *Compute) Execute(ctx context.Context, request capabilities.CapabilityRe
 	return c.executeWithModule(m.module, config, request)
 }
 
-func (c *Compute) initModule(id string, binary []byte, workflowID, referenceID string) (*module, error) {
+func (c *Compute) initModule(id string, binary []byte, workflowID, workflowExecutionID, referenceID string) (*module, error) {
 	initStart := time.Now()
-	mod, err := host.NewModule(&host.ModuleConfig{Logger: c.log}, binary)
+	mod, err := host.NewModule(&host.ModuleConfig{
+		Logger: c.log,
+		Fetch: func(req *wasmpb.FetchRequest) (*wasmpb.FetchResponse, error) {
```

Comment:
Could we extract this out into a wrapper over outgoingConnectorHandler?

I know we've (specifically I) used inlining in the past, but our functions are getting too big for that and would benefit from being abstracted out.